jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java

```diff
@@ -100,15 +100,23 @@ static String getUserAgent() {
 
   @Nullable private final Authorization authorization;
   private final RegistryEndpointProperties registryEndpointProperties;
+  private final boolean allowHttp;
 
   /**
    * @param authorization the {@link Authorization} to access the registry/repository
    * @param serverUrl the server URL for the registry (for example, {@code gcr.io})
    * @param imageName the image/repository name (also known as, namespace)
+   * @param allowHttp if {@code true}, allows redirects and fallbacks to HTTP; otherwise, only
+   *     allows HTTPS
    */
-  public RegistryClient(@Nullable Authorization authorization, String serverUrl, String imageName) {
+  public RegistryClient(
+      @Nullable Authorization authorization,
+      String serverUrl,
+      String imageName,
+      boolean allowHttp) {
```

Comment:
I think there's something to readability when using builders or static factories here. Having to look at what a "true/false" parameter means when reading the code is not the most readable experience.

Reply:
Added static factories.